sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -885,6 +892,14 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog with QueryErrorsB
                 "expr" -> variantExpr.sql,
                 "dataType" -> toSQLType(variantExpr.dataType)))
 
+          case o if mapExprInPartitionExpression(o).isDefined =>
+            val mapExpr = mapExprInPartitionExpression(o).get
+            o.failAnalysis(
+              errorClass = "UNSUPPORTED_FEATURE.PARTITION_BY_MAP",
+              messageParameters = Map(
+                "expr" -> mapExpr.sql,
```

Comment:
Please, quite to `toSQLExpr`

Reply:
Good point! I adjusted `case o if variantExprInPartitionExpression(o).isDefined =>` to reuse `toSQLExpr` as well. Unit tests are adjusted accordingly!